packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/tasks/VideoResultTask.java

```diff
@@ -0,0 +1,75 @@
+package expo.modules.imagepicker.tasks;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.os.Bundle;
+
+import org.unimodules.core.Promise;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import androidx.annotation.NonNull;
+import expo.modules.imagepicker.ImagePickerConstance;
+import expo.modules.imagepicker.ImagePickerFileUtils;
+
+public class VideoResultTask extends ImagePickerResultTask {
```

Comment:
I see why you decided to go with inheritance here. It is good enough for me, but we might spend some time to try and refactor it if you wish.

Reply:
So you want to change it to a composition? or you suggest a different approach here?